openedx/core/djangoapps/user_authn/api/tests/test_views.py

```diff
@@ -0,0 +1,169 @@
+            },
+        ]
+
+    def get_context(self, params=None, current_provider=None, backend_name=None, add_user_details=False):
+        """
+        Returns the third party auth context
+        """
+        return {
+            'currentProvider': current_provider,
+            'providers': self.get_provider_data(params) if params else [],
+            'secondaryProviders': [],
+            'finishAuthUrl': pipeline.get_complete_url(backend_name) if backend_name else None,
+            'errorMessage': None,
+            'registerFormSubmitButtonText': 'Create Account',
+            'syncLearnerProfileData': False,
+            'pipeline_user_details': {'email': 'test@test.com'} if add_user_details else {}
+        }
+
+    def test_missing_arguments(self):
+        """
+        Test that if required arguments are missing, proper status code and message
+        is returned.
+        """
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.data, {'message': 'Request missing required parameter: redirect_to'})
+
+    @patch.dict(settings.FEATURES, {'ENABLE_THIRD_PARTY_AUTH': False})
+    def test_no_third_party_auth_providers(self):
+        """
+        Test that if third party auth is enabled, context returned by API contains
+        the provider information
+        """
+        response = self.client.get(self.url, self.query_params)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data, self.get_context())
+
+    def test_third_party_auth_providers(self):
+        """
+        Test that api returns details of currently enabled third party auth providers
+        """
+        response = self.client.get(self.url, self.query_params)
+        params = {
+            'next': self.query_params['redirect_to']
+        }
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data, self.get_context(params))
+
+    @ddt.data(
+        ('google-oauth2', 'Google', False),
+        ('facebook', 'Facebook', False),
+        ('google-oauth2', 'Google', True)
+    )
+    @ddt.unpack
+    def test_running_pipeline(self, current_backend, current_provider, add_user_details):
+        """
+        Test that when third party pipeline is running, the api returns details
+        of current provider
+        """
+        email = 'test@test.com' if add_user_details else None
+        params = {
+            'next': self.query_params['redirect_to']
+        }
+
+        # Simulate a running pipeline
+        if current_backend:
+            pipeline_target = 'openedx.core.djangoapps.user_authn.views.login_form.third_party_auth.pipeline'
+            with simulate_running_pipeline(pipeline_target, current_backend, email=email):
+                response = self.client.get(self.url, self.query_params)
+        else:
```

Comment:
When this else statement is going to run? I believe the `if current_backend` will always be `True` for the data you provided to this test.

Reply:
I will update the test